File: frontend/src/types/task.ts
export interface Task {
  id: number;
  titulo: string;
  descricao?: string;
  status: StatusTask;
  dataCriacao: string;
}

export interface TaskDTO {
  titulo: string;
  descricao?: string;
  status: StatusTask;
}

export enum StatusTask {
  NAO_INICIADO = 'NAO_INICIADO',
  EM_ANDAMENTO = 'EM_ANDAMENTO',
  CONCLUIDO = 'CONCLUIDO'
}

export const StatusTaskLabels = {
  [StatusTask.NAO_INICIADO]: 'Não Iniciado',
  [StatusTask.EM_ANDAMENTO]: 'Em Andamento',
  [StatusTask.CONCLUIDO]: 'Concluído'
};

export const StatusTaskColors = {
  [StatusTask.NAO_INICIADO]: 'bg-gray-100 text-gray-800 border-gray-300',
  [StatusTask.EM_ANDAMENTO]: 'bg-blue-100 text-blue-800 border-blue-300',
  [StatusTask.CONCLUIDO]: 'bg-green-100 text-green-800 border-green-300'
};

File: backend/src/main/java/com/backend/model/Task.java
package com.backend.model;

import com.fasterxml.jackson.annotation.JsonFormat;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "tasks")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Task {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(nullable = false)
    private String titulo;
    
    @Column(columnDefinition = "TEXT")
    private String descricao;
    
    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private StatusTask status = StatusTask.NAO_INICIADO;
    
    @CreationTimestamp
    @Column(name = "data_criacao", nullable = false, updatable = false)
    @JsonFormat(pattern = "dd-MM-yyyy HH:mm:ss")
    private LocalDateTime dataCriacao;
    
    public enum StatusTask {
        NAO_INICIADO("Não Iniciado"),
        EM_ANDAMENTO("Em Andamento"),
        CONCLUIDO("Concluído");
        
        private final String descricao;
        
        StatusTask(String descricao) {
            this.descricao = descricao;
        }
        
        public String getDescricao() {
            return descricao;
        }
    }
}